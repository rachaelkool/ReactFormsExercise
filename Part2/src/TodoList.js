import React, {useState} from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";


const TodoList = () => {
    const [todos, setTodos] = useState([]);
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, newTodo]);
    };
    const removeTodo = id => {
        setTodos(todos => todos.filter(todo => todo.id !== id));
    };

  const todoComponents = todos.map(todo => (
    <Todo
        key={todo.id}
        id={todo.id}
        task={todo.task}
        handleRemove={removeTodo}
    />
  ));

  return (
    <div>
      <NewTodoForm addTodo={addTodo} />
      <ul>{todoComponents}</ul>
    </div>
  );
}

export default TodoList;