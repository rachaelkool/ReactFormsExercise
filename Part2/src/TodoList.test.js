import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";


it('renders without crashing', function() {
    render(<TodoList />);
});

it('matches snapshot', function() {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it('can add a todo', function() {
    const {getByText, getByLabelText} = render(<TodoList />);
  
    const taskInput = getByLabelText("Task:");
    fireEvent.change(taskInput, { target: { value: 'feed my dog' } });
    const addBtn = getByText('Add a new todo!');
    fireEvent.click(addBtn);
  
    const removeBtn = getByText("X");

    expect(removeBtn).toBeInTheDocument();
    expect(taskInput.value).toBe('');
});

it('can remove a todo', function() {
    const {getByText, getByLabelText} = render(<TodoList />);
  
    const taskInput = getByLabelText("Task:");
    fireEvent.change(taskInput, { target: { value: 'feed my dog' } });
    const addBtn = getByText('Add a new todo!');
    fireEvent.click(addBtn);
  
    const removeBtn = getByText("X");
    fireEvent.click(removeBtn);

    expect(removeBtn).not.toBeInTheDocument();
    expect(taskInput.value).toBe('');
});