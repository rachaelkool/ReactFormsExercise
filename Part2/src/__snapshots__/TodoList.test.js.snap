// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <label
          for="task"
        >
          Task:
        </label>
        <input
          id="task"
          name="task"
          placeholder="task"
          type="text"
          value=""
        />
        <button>
          Add a new todo!
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
